lib/cli/src/commands/check/check-service.ts

```diff
@@ -27,25 +27,15 @@ export class DefaultCheckService {
    *   4) Checking that matching routes are compatible
    */
   async checkSpec(specPath: string, options: CheckSpecCommandOptions): Promise<void> {
-    // Read the spec file as raw YAML/JSON first
-    const specContent = fs.readFileSync(specPath, "utf8");
-    const rawSpec = yaml.load(specContent) as OpenAPISchema;
-
-    // Convert OpenAPI v3.1 to v3.0 if needed
-    const convertedSpec = convertOpenApiToV3(rawSpec) as OpenAPIV3.Document;
```

Comment:
Moves this conversion logic to a shared function since CLI users can pass in their own base spec